engine/access/ingestion2/results_forest.go

```diff
@@ -614,21 +696,32 @@ func (rf *ResultsForest) getOrCreateContainer(result *flow.ExecutionResult, bloc
 		return nil, fmt.Errorf("failed to store receipt's container: %w", err)
 	}
 	rf.forest.AddVertex(container)
-
-	switch blockStatus {
-	case BlockStatusSealed:
-		if err := container.SetBlockStatus(blockStatus); err != nil {
-			return nil, fmt.Errorf("failed to set block status to sealed: %w", err)
+	// TODO: as mentioned in the context of function `ensureAdmissible`, I think it would be beneficial to
+	//       move the LevelledForest operation (above) inside our function `ensureAdmissible`, toge
+	//       However, this will require additional refactoring, to move modifications of the ResultStatus above
+	//       outside the scope of `getOrCreateContainer`. I think conceptually we are holding the lock `rf.mu` here
+	//       because we want to potentially add a container. We are done with that now. I think it would improve
+	//       code clarity and robustness if subsequent modifications of the container would be handled by a separate
+	//       method.
+	//       It should be noted that, for some updates of ResultStatus, we operate solely on the container may not require
+	//       holding the lock `rf.mu`. While this is of no performance concern, getting vertices from the forest is
+	//       still conceptually different than operating on the vertices itself. In other words, there are similar usage
+	//       patterns, but conceptually different operations.
```

Comment:
I'd like to specifically draw attention to this TODO, which I think would further improve clarity significantly.